handshake/crypto_setup_server.go

```diff
@@ -152,6 +153,9 @@ func (h *cryptoSetupServer) handleMessage(chloData []byte, cryptoData map[Tag][]
 		if err != nil {
 			return false, err
 		}
+		h.mutex.Lock()
+		h.wroteSHLO = true
```

Comment:
Isn't there a race condition here, as the frame may already be packed (and Write() returned) before the sealer is actually requested?